### ISSUE TITLE
Differences in variable importance when passing a SDMmodelCV object compared to passing each of the models inside the SDMmodelCV object

### ISSUE THREAD

**Issue by veroandreo** [ConsBiol-unibern/SDMtune#4]
Hello, 

I'm training a maxnet model with CV to model the occurrence of a disease. I have noticed an important difference in variables importance when I pass the SDMmodelCV object in comparison to passing each of the models inside the SDMmodelCV object to the function `varImp`

Here the commands and results:
```
# crossvalidation with train data ( AUC value will be averaged across the different models )
model <- train("Maxnet", p = presence, a = bg_model, fc = "lh", reg = 0.5, seed = 37, rep = 3)

# permutation importance
vi_maxnet <- varImp(model)
vi_maxnet

                           Variable Permutation_importance    sd
1         LC08_2014_2015_cva_change                   17.6 0.001
2  LC08_2014_2015_cva_change_dist_2                   10.5 0.001
3  LC08_2014_2015_cva_change_dist_1                    9.9 0.001
4                 rf_class_20140719                    9.9 0.001
5  LC08_2014_2015_cva_change_dist_4                    9.6 0.000
6  LC08_2014_2015_cva_change_dist_3                    9.2 0.000
7          LC08_2014_2015_cva_angle                    8.6 0.000
8           LC08_20140719_tasscap.3                    8.4 0.000
9           LC08_20140719_tasscap.2                    8.3 0.000
10          LC08_20140719_tasscap.4                    8.0 0.001
```

Here, the results of passing each of the 3 models in the object:

```
varImp(model@models[[1]])
                           Variable Permutation_importance    sd
1  LC08_2014_2015_cva_change_dist_4                   28.8 0.024
2           LC08_20140719_tasscap.4                   15.9 0.035
3  LC08_2014_2015_cva_change_dist_2                   15.5 0.019
4  LC08_2014_2015_cva_change_dist_1                   15.4 0.025
5                 rf_class_20140719                    8.1 0.014
6         LC08_2014_2015_cva_change                    8.0 0.014
7           LC08_20140719_tasscap.2                    3.4 0.005
8          LC08_2014_2015_cva_angle                    2.4 0.006
9  LC08_2014_2015_cva_change_dist_3                    2.2 0.007
10          LC08_20140719_tasscap.3                    0.5 0.002

varImp(model@models[[2]])
                           Variable Permutation_importance    sd
1  LC08_2014_2015_cva_change_dist_4                   40.8 0.023
2  LC08_2014_2015_cva_change_dist_1                   16.2 0.031
3  LC08_2014_2015_cva_change_dist_2                   15.3 0.026
4                 rf_class_20140719                    7.7 0.011
5           LC08_20140719_tasscap.2                    7.4 0.018
6         LC08_2014_2015_cva_change                    4.4 0.011
7           LC08_20140719_tasscap.4                    3.8 0.014
8  LC08_2014_2015_cva_change_dist_3                    3.0 0.005
9          LC08_2014_2015_cva_angle                    0.7 0.004
10          LC08_20140719_tasscap.3                    0.6 0.004

varImp(model@models[[3]])
                           Variable Permutation_importance    sd
1  LC08_2014_2015_cva_change_dist_4                   53.4 0.030
2  LC08_2014_2015_cva_change_dist_1                   15.5 0.020
3  LC08_2014_2015_cva_change_dist_2                   13.1 0.015
4           LC08_20140719_tasscap.2                    5.6 0.009
5                 rf_class_20140719                    5.4 0.007
6           LC08_20140719_tasscap.4                    3.4 0.009
7          LC08_2014_2015_cva_angle                    3.1 0.006
8  LC08_2014_2015_cva_change_dist_3                    0.5 0.001
9           LC08_20140719_tasscap.3                    0.1 0.000
10        LC08_2014_2015_cva_change                    0.0 0.000
```

The result of the first command and the importances obtained do not seem to be an average (as I would expect) of the importance of those variables within each of the models in the SDMmodelCV object. How are they estimated and how should they be interpreted then? Any hints are more than welcome.

Thanks in advance

**Comment by sgvignali** [comment]
Hi, the varImp function accepts only SDMmodel objects, and not SDMmodelCV objects, as you can see from the documentation.
I should add an error when using the function with an SDMmodelCV object, thanks for that. And actually that means that "varSel" and "reduceVar" should not be used with an "SDMmodelCV" object.

I'll try to implement the varImp function for cross validation as soon as I have time, but for the moment I will add the error message!

Thanks for reporting that

**Comment by sgvignali** [comment]
Ok I implemented the function, you can download the github version and test it.
Please note that the value of the variable importance of a SDMtuneCV object might be a bit different from the average of the variable importance of each single model. 
This is because the seeds are set only once for all the permutations, while when you call varImp for the a single model, the seeds are reset at each call.

**Comment by veroandreo** [comment]
Hi, 

Thanks a lot for this! 

I just tested and as you say it's not precisely the average, but at least it is in "range" now. 

It is then possible now to use a SDMmodelCV object with `varSel` and `reduceVar`? The latter was not issuing any error before the change in `varImp` anyway. I wonder what was it doing then.

**Comment by sgvignali** [comment]
Yes, now you can use `varSel` and `reduceVar` with SDMmodelCV objects! Then I close this issue

**Comment by sgvignali** [comment]
Fixed with afe7fd884b619439c1371cb5f7369e2269e001d2
